### PR TITLE
feat(tests): bump test coverage for frontend files

### LINKED ISSUE #693: (SP: 5) [Frontend] Bump test coverage and resolve warnings for admin panel
## Description

To maintain code health and prevent regression during active development on the admin panel, we need to address accumulated test gaps and console warnings.

This task focuses exclusively on the frontend (client) part of the admin panel. Splitting this from the backend will help us deliver smaller, cleaner PRs and minimize potential merge conflicts while other team members are actively working on the admin features.

Our main goals are to eliminate files with 0% coverage, significantly bump overall coverage, and clean up test runner warnings.

## How to reproduce / How to run

1. Navigate to the admin rep
2. Run the test suite with the coverage flag: `npm run test`
3. Check the generated coverage report to locate files with` 0%` or low coverage.
4. Watch the terminal output during test execution to identify active warnings/deprecation logs.

## Acceptance Criteria

- No 0% covered files: All frontend files in the admin panel must have at least some test coverage (no more "0 coverage" files).
- High Coverage Target: Overall test coverage for the admin frontend is bumped as high as possible, aiming for 95%+ (or a significant step towards it).
- Zero Warnings: All test-related console warnings, deprecation logs, or React act-wrapper warnings are fully resolved.
- No Merge Conflicts: Changes are scoped tightly to minimize conflicts with other active admin tasks.

### PR DESCRIPTION
**Pull request by dest411**
## Description

This PR significantly improves code health and test coverage for the frontend (client-side) part of the admin panel located in the `app/` directory.

### Key Improvements:
- **Coverage Bump:** Raised unit test coverage for `app/` frontend modules, bringing the average client-side test coverage close to **96%**.
- **Console Cleanup:** Resolved all test runner console warnings, including React `act(...)` state update warnings, missing `key` prop warnings, invalid DOM property logs, and Apollo Client deprecation notices.
- **Code Quality:** Addressed SonarQube issues (parameterized duplicate tests via `it.each`, eliminated duplicate test titles) and strictly typed all mock data without using `any`.

> **Note:** A few files currently remain around 80–85% coverage on specific metrics. These will be covered in the upcoming PR dedicated to backend coverage and the remaining files.

## Type of change

- [ ] New feature
- [x] Bug fix
- [ ] Documentation update
- [x] Refactoring / Tech debt
- [ ] CI/CD improvement
- [ ] Other: 

## How to test

1. Navigate to the root of the repository.
2. Run the test suite:
   ```bash
   npm run test
3. Verify that test files located inside the app/ directory execute successfully.
4. Observe the terminal output during test execution to confirm that no act(...), key, or console warnings/errors appear.

## Expected result:
- All Tests Passing: 100% of unit tests pass cleanly without errors.
- Zero Console Warnings: Terminal output is clean with 0 console warnings or deprecation logs during test execution.
- High Coverage Target: Average frontend test coverage for app/ files reaches ~96%.
- Clean Git State: Branch is cleanly rebased onto develop and passes pre-commit linter checks.

## Checklist

- [x] Code compiles and runs correctly
- [x] Tests pass successfully
- [x] Linting checks are clean
- [x] No Sonar issues
- [ ] Documentation updated (if applicable)
- [x] All comments and requested changes addressed
- [x] Branch is up to date with the target branch
- [x] Linked issue/task included
- [x] Task moved to the review column on the board

Closes: https://github.com/Liatoshynsky-Foundation/lf-admin/issues/693
---

### CODE REVIEW

File: app/lib/parser/parserUtils.ts [hunk 1]
```diff
@@ -55,30 +50,29 @@ export function parseJsonLd(html: string): Record<string, unknown> | null {
...
+  const m = /^(\d{1,2})\.(\d{1,2})\.(\d{4})(?:[ T](\d{1,2}):(\d{2}))?/.exec(trimmed);
 
   if (m) {
     const day = Number.parseInt(m[1], 10);
     const month = Number.parseInt(m[2], 10) - 1;
     const year = Number.parseInt(m[3], 10);
     const hour = m[4] ? Number.parseInt(m[4], 10) : 0;
     const min = m[5] ? Number.parseInt(m[5], 10) : 0;
-    const dt = new Date(Date.UTC(year, month, day, hour, min, 0));
+    const dt = new Date(year, month, day, hour, min);
```
Comment (Halyna-Trush) [inline app/lib/parser/parserUtils.ts]: Great work!

Using `new Date(...)` here makes the parsed value depend on the server’s timezone. For example, `15.01.2020 05:06` becomes `02:06Z` when parsed in UTC+3, but remains `05:06Z` when parsed in UTC.
Since `ParsePublishedDate` later calls `toISOString()`, the same input could produce different stored values in different environments. The test using `getHours()` does not catch this because it checks the local hour.
If this value is meant to be UTC, please keep `Date.UTC(...)` and assert against `toISOString()`. If it represents Kyiv time, the timezone should be specified explicitly instead of relying on the server configuration.

File: app/shared/components/side-navigation/collapse-list-navigation/CollapseListNavigation.tsx [hunk 2]
```diff
@@ -62,12 +62,7 @@ export const CollapseListNavigation: React.FC<CollapseListNavigationProps> = ({
 
   return (
     <Box sx={{ position: 'relative' }} onMouseEnter={handleMouseEnter} onMouseLeave={handleMouseLeave}>
-      <ListElement
-        element={element}
-        open={openNavbar}
-        handleClick={openNavbar ? handleClick : undefined}
-        sxItem={{ mb: '0' }}
-      >
+      <ListElement element={element} open={openNavbar} handleClick={handleClick} sxItem={{ mb: '0' }}>
```
Comment (Halyna-Trush) [inline app/shared/components/side-navigation/collapse-list-navigation/CollapseListNavigation.tsx:68]: Clicking the parent item in the collapsed state appears to be intentional, but `anchorEl` is still only set in `handleMouseEnter`. On touch devices or when using the keyboard, the click may open the submenu while `anchorEl` is still `null`, which can cause the `Popper` to be positioned incorrectly.
Could we set `anchorEl` in `handleClick` as well? It would also be useful to cover this case in the test, since the current `Popper` mock ignores `anchorEl`.

File: app/lib/parser/parserUtils.ts (hunk 1 above)
Comment (dest411) [reply app/lib/parser/parserUtils.ts]: Yeah, you're right!

File: app/shared/components/side-navigation/collapse-list-navigation/CollapseListNavigation.tsx (hunk 2 above)
Comment (dest411) [reply app/shared/components/side-navigation/collapse-list-navigation/CollapseListNavigation.tsx:68]: Good catch!

**Review by Halyna-Trush — APPROVED** [review]

**Review by DenisGordProgrammer — APPROVED** [review]
Approved. This is a strong test-focused improvement: it adds meaningful branch and edge-case coverage across the frontend, improves mock quality, and keeps the test suite and Sonar checks clean. This will make future UI changes safer and reduce regressions in shared admin flows.

**Comment by DenisGordProgrammer** [comment]
Nice work on this one. You put in a lot of solid effort here, and the extra coverage should make the admin frontend much safer to change. Respect for the thorough test pass.
